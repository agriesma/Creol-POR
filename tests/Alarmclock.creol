-- CVM cvm "viisi.ifi.uio.no"
-- startobject (cvm1, Alarmclock, null)  

class Alarmclock
 begin
 	var now: Int

 op run == now := 0

 with Any
  op wakeme(in n: Int) == var alarmsetting: Int ;
 		alarmsetting := now + n;
 		await now >= alarmsetting
 	op tick == now := now + 1
 end
