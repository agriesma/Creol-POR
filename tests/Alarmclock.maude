load interpreter
mod PROGRAM is
pr INTERPRETER .
op init : -> Configuration [ctor] .
eq init =
< 'Alarmclock : Cl | Inh: noInh, Par: noAid, Att: 'now |-> null, Mtds: 
  < 'run : Mtdname | Param: noAid, Latt: noSubst, Code: 'now ::= int(0) ; return ( emp ) > *
  < 'wakeme : Mtdname | Param: 'n, Latt: 'n |-> null , 'alarmsetting |-> null, Code: 'alarmsetting ::= ( 'plus[[ 'now # 'n ]] ) ; ( await ( 'lessEq[[ 'alarmsetting # 'now ]] ) ) ; return ( emp ) > *
  < 'tick : Mtdname | Param: noAid, Latt: noSubst, Code: 'now ::= ( 'plus[[ 'now # int(1) ]] ) ; return ( emp ) >, Ocnt: 0 >

.
endm
