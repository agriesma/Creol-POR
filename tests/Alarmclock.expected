==========================================
reduce in PROGRAM : init .
result Class: 
< 'Alarmclock : Cl | Inh: noInh,Par: noAid,Att: 'now |-> null,Mtds:
< 'run : Mtdname | Param: noAid,Latt: noSubst,Code: ('now assign int(0)) ;
    return(emp) > *
< 'tick : Mtdname | Param: noAid,Latt: noSubst,Code: 'now ::= ('plus[['now #
    int(1)]]) ; return(emp) > *
< 'wakeme : Mtdname | Param: 'n,Latt: 'alarmsetting |-> null, 'n |-> null,Code:
    'alarmsetting ::= ('plus[['now # 'n]]) ; await ('lessEq[['alarmsetting #
    'now]]) ; return(emp) >,Ocnt: 0 >

Bye.
